From akinobu.mita@gmail.com Sun Feb 17 01:21:51 2008
Return-Path: <akinobu.mita@gmail.com>
Received: from imap.linux.ibm.com ([unix socket]) by imap.linux.ibm.com
	(Cyrus v2.3.7-Invoca-RPM-2.3.7-7) with LMTPA; Sun, 17 Feb 2008 01:21:51
	-0500
X-Sieve: CMU Sieve 2.3
Received: by imap.linux.ibm.com (Postfix, from userid 101) id 9A91D1910144;
	Sun, 17 Feb 2008 01:21:51 -0500 (EST)
X-Spam-TestScore: SPF_NEUTRAL=1.379
X-Spam-TokenSummary: Bayes not run.
X-Spam-Checker-Version: SpamAssassin 3.1.7 (2006-10-05) on
	imap.linux.ibm.com
X-Spam-Level: *
X-Spam-Status: No, score=1.4 required=5.0 tests=SPF_NEUTRAL 
	autolearn=disabled version=3.1.7
X-Spam-Relay-Country: US ** US US US ** US US XX JP
Received: from smtp.linux.ibm.com (smtp.linux.ibm.com [9.26.4.197]) by
	imap.linux.ibm.com (Postfix) with ESMTP id 1CE5B1910121 for
	<cmm@imap.linux.ibm.com>; Sun, 17 Feb 2008 01:21:47 -0500 (EST)
Received: from localhost (localhost.localdomain [127.0.0.1]) by
	smtp.linux.ibm.com (Postfix) with ESMTP id CB34CC03E for
	<cmm@linux.ibm.com>; Sun, 17 Feb 2008 01:21:46 -0500 (EST)
X-Virus-Scanned: amavisd-new at linux.ibm.com
Received: from BLDVMB.POK.IBM.COM (bldvmb.pok.ibm.com [9.57.6.113]) by
	smtp.linux.ibm.com (Postfix) with ESMTP id 7F377C03F for
	<cmm@linux.ibm.com>; Sun, 17 Feb 2008 01:21:46 -0500 (EST)
Received:  by BLDVMB.POK.IBM.COM (IBM VM SMTP Level 440) via spool with
	SMTP id 2237 ; Sat, 16 Feb 2008 23:22:47 MST
Received: by bldvmb.vnet.ibm.com (xagent2 5.9.5) via xagsmtp2 with spool id
	2472 for cmm@linux.vnet.ibm.com; Sat, 16 Feb 2008 23:22:47 -0700 (MST)
Received: from d03relay05.boulder.ibm.com [9.17.195.107] by
	BLDVMB.POK.IBM.COM (IBM VM SMTP Level 440) via TCP with ESMTP ; Sat, 16 Feb
	2008 23:22:47 MST
Received: from d03av04.boulder.ibm.com (d03av04.boulder.ibm.com
	[9.17.195.170]) by d03relay05.boulder.ibm.com (8.13.8/8.13.8/NCO v8.7) with
	ESMTP id m1H6LiXO136774 for <cmm@us.ibm.com>; Sat, 16 Feb 2008 23:21:45
	-0700
Received: from d03av04.boulder.ibm.com (loopback [127.0.0.1]) by
	d03av04.boulder.ibm.com (8.12.11.20060308/8.13.3) with ESMTP id
	m1H6Likx030750 for <cmm@us.ibm.com>; Sat, 16 Feb 2008 23:21:44 -0700
Received: from d03as06.boulder.ibm.com (d03as06 [9.17.195.76]) by
	d03av04.boulder.ibm.com (8.12.11.20060308/8.12.11) with ESMTP id
	m1H6LiVl030744 (version=TLSv1/SSLv3 cipher=DHE-RSA-AES256-SHA bits=256
	verify=OK) for <cmm@us.ibm.com>; Sat, 16 Feb 2008 23:21:44 -0700
Received: from e32.co.us.ibm.com ([9.17.249.42]) by d03as06.boulder.ibm.com
	(8.12.11.20060308/8.12.11) with ESMTP id m1H697Cp023346
	(version=TLSv1/SSLv3 cipher=DHE-RSA-AES256-SHA bits=256 verify=OK) for
	<cmm@us.ibm.com>; Sat, 16 Feb 2008 23:09:07 -0700
Received: from rv-out-0910.google.com (rv-out-0910.google.com
	[209.85.198.191]) by e32.co.us.ibm.com (8.13.8/8.13.8) with ESMTP id
	m1H6LWHg002092 for <cmm@us.ibm.com>; Sun, 17 Feb 2008 01:21:35 -0500
Received: by rv-out-0910.google.com with SMTP id l15so1242892rvb.44 for
	<cmm@us.ibm.com>; Sat, 16 Feb 2008 22:21:37 -0800 (PST)
DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed;        d=gmail.com;
	s=gamma;
	h=domainkey-signature:received:received:date:from:to:cc:subject:message-id:mime-version:content-type:content-disposition:user-agent;
	       bh=BEUNb5fbU7TGkau+2NZhsb0BFNMSfr0Csn4RjVOWzNE=;       
	b=bomj0xNICc6yAza+b2DN8jMdBOjz+0Zinq5Mlqv+dno4xAtQCp8aGTNabVKexhDxyXCDUXbXcHS9yOtZc7muuXJZ6uQdxZnWFTZhTl6MzzYC2RoQ456zCLJglu2XIr/6bcD6DG6Oy8wLDspTK4yjlxWDIvHz01Bh0Bcy8GNcFpw=
DomainKey-Signature: a=rsa-sha1; c=nofws;        d=gmail.com; s=gamma;
	h=date:from:to:cc:subject:message-id:mime-version:content-type:content-disposition:user-agent;
	      
	b=YO3mwdDJdIlbNYtyCduojnprKkerUhloqxrlk1YKzrWEc3OgAuv4Ia7T03bmJKGmGf1iVMobRhxAJmwB3ZzxymHhW/XD8cv1myzKwqPSXdMag3ODkr1xrRXRQWEE3oxdx/XAJe5OOkDUSmqQB0Rhc3T+Wvqp46Y4IoCZV6MHfzI=
Received: by 10.140.158.4 with SMTP id g4mr2993143rve.44.1203229297682;
	Sat, 16 Feb 2008 22:21:37 -0800 (PST)
Received: from @ ( [222.148.118.78]) by mx.google.com with ESMTPS id
	k34sm5957763rvb.23.2008.02.16.22.21.35 (version=SSLv3 cipher=OTHER); Sat,
	16 Feb 2008 22:21:37 -0800 (PST)
Date: Sun, 17 Feb 2008 15:15:20 +0900
From: Akinobu Mita <akinobu.mita@gmail.com>
To: linux-ext4@vger.kernel.org
Cc: Andrew Morton <akpm@linux-foundation.org>,        Stephen Tweedie <sct@redhat.com>, adilger@clusterfs.com,        Mingming Cao <cmm@us.ibm.com>, Theodore Tso <tytso@MIT.EDU>
Subject: [PATCH 4/4] ext4: check ext4_journal_get_write_access() errors
Message-ID: <20080217061519.GK3390@APFDCB5C>
MIME-Version: 1.0
Content-Type: text/plain; charset=iso-2022-jp
Content-Disposition: inline
User-Agent: Mutt/1.5.17 (2007-11-01)
X-Xagent-From: akinobu.mita@gmail.com
X-Xagent-To: cmm@linux.vnet.ibm.com
X-Xagent-Gateway: bldvmb.vnet.ibm.com (XAGENTU at BLDVMB)
X-Evolution-Source: imap://cmm@imap.linux.ibm.com/
Content-Transfer-Encoding: 8bit

Check ext4_journal_get_write_access() errors.

Signed-off-by: Akinobu Mita <akinobu.mita@gmail.com>
Cc: Stephen Tweedie <sct@redhat.com>
Cc: adilger@clusterfs.com
Cc: Andrew Morton <akpm@linux-foundation.org>
Cc: Mingming Cao <cmm@us.ibm.com>
Cc: Theodore Tso <tytso@MIT.EDU>
---
 fs/ext4/namei.c |    9 +++++++--
 1 file changed, 7 insertions(+), 2 deletions(-)

Index: 2.6-rc/fs/ext4/namei.c
===================================================================
--- 2.6-rc.orig/fs/ext4/namei.c
+++ 2.6-rc/fs/ext4/namei.c
@@ -57,10 +57,15 @@ static struct buffer_head *ext4_append(h

 	*block = inode->i_size >> inode->i_sb->s_blocksize_bits;

-	if ((bh = ext4_bread(handle, inode, *block, 1, err))) {
+	bh = ext4_bread(handle, inode, *block, 1, err);
+	if (bh) {
 		inode->i_size += inode->i_sb->s_blocksize;
 		EXT4_I(inode)->i_disksize = inode->i_size;
-		ext4_journal_get_write_access(handle,bh);
+		*err = ext4_journal_get_write_access(handle, bh);
+		if (*err) {
+			brelse(bh);
+			bh = NULL;
+		}
 	}
 	return bh;
 }
